floater/generators.py

```diff
@@ -88,15 +103,24 @@ def get_rectmesh(self):
         RETURNS
         -------
         x : np.ndarray
-            2D array of float x coordinates
+            3D array of float x coordinates
         y : np.ndarray
-            2D array of float y coordinates
+            3D array of float y coordinates
+        z : np.ndarray
+            3D array of float z coordinates
         """
 
-        xx, yy = np.meshgrid(self.x, self.y)
-        if self.model_grid is not None:
-            xx, yy = self.subset_floats_from_mask(xx, yy)
-        return xx, yy
+        if np.all(self.z==-0.5):
```

Comment:
I would add some comments to explain this logic. I can't really follow what is happening and why you are making this check on `self.z`.